sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -44,6 +44,14 @@ trait SQLQueryTestHelper {
       .replaceAll("Last Access.*", s"Last Access $notIncludedMsg")
       .replaceAll("Partition Statistics\t\\d+", s"Partition Statistics\t$notIncludedMsg")
       .replaceAll("\\*\\(\\d+\\) ", "*") // remove the WholeStageCodegen codegenStageIds
+
+      // Below is needed since the catalog table in LogicalRelation can produce serde class
+      // "optionally" if CatalogTable is presented and has a serde information in the storage.
+      // This assumes LogicalRelation contains a catalog table, otherwise it would not match with
+      // this pattern.
+      .replaceAll("Arguments: (?<relation>.+), (?<attr>\\[(.+#x)(, .+#x)?\\]), " +
```

Comment:
This still looks hacky to me. Can we have a special config to exclude serde info in the explain string and set this config in `SQLQueryTestSuite`?

Reply:
I'm OK either way. I feel odd to have a config to just satisfy the test code, but if it leads the code simpler and makes everyone be happy, why not? I'll reflect the suggestion.